### ISSUE TITLE
Error in generated Dart client code: Undefined name 'OneOf'. 

### ISSUE THREAD

**Issue by robinbakkerus** [OpenAPITools/openapi-generator#3884]
The generated Dart code gives the error:  'Undefined name 'OneOf' on this line in klantbeeld.dart:
OneOf&lt;VroegPensioenAanspraak,WeduwePensioenAanspraak&gt;.listFromJson(json['aanspraken']);

The command used to generate the Dart code:
`java -jar modules\openapi-generator-cli\target\openapi-generator-cli.jar generate -g dart -i klantbeeld.yaml -o  /tmp`

Using the openapi-generator (https://github.com/OpenAPITools/openapi-generator) version: 4.1.3-SNAPSHOT

The input swagger file 
```

openapi: 3.0.0
servers:
  - url: 'http://localhost:4567/'

info:
  description: generated api for GetKlantbeeld
  version: 1.0.0
  title: GetKlantbeeld
 
paths:
  /GetKlantbeeld:
    post:
      summary: 'TODO'
      description: 'TODO'
      operationId: GetKlantbeeld
      responses:
        '200':
          description: OK
          content:
            application/json:
              schema:
                $ref: '#/components/schemas/Klantbeeld'
        '500':
          description: Unknown error encountered
      requestBody:
        required: true
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/SelectKlantbeeld'

components:
  schemas:
  
    PensioenAanspraak:
      type: object
      properties:
        aanspraakType:
          $ref: '#/components/schemas/PensioenAanspraakType'
        uitkering:
          type: number
        einddatum:
          type: string
        _classname:
          type: string
    
    VroegPensioenAanspraak:
      allOf:
        - $ref: "#/components/schemas/PensioenAanspraak"
      type: object
      properties:
        looptijd:
          type: number
    
    WeduwePensioenAanspraak:
      allOf:
        - $ref: "#/components/schemas/PensioenAanspraak"
      type: object
      properties:
        partnerOverleden:
          type: string
    
    Adres:
      type: object
      properties:
        adres:
          type: string
        postcode:
          type: string
        plaats:
          type: string
  
    Geslacht:
      type: string
      enum:
        - MAN
        - VROUW
  
    SelectKlantbeeld:
      type: object
      properties:
        klantId:
          type: integer
        achternaam:
          type: string
  
    Klantbeeld:
      type: object
      properties:
        klantId:
          type: integer
        voornaam:
          type: string
        achternaam:
          type: string
        geboorteDatum:
          type: string
        geslacht:
          $ref: '#/components/schemas/Geslacht'
        saldo:
          type: number
        adres:
          $ref: '#/components/schemas/Adres'
        aanspraken:
          type: array
          items:
            oneOf:
              - $ref: '#/components/schemas/VroegPensioenAanspraak'
              - $ref: '#/components/schemas/WeduwePensioenAanspraak'
            discriminator:
              propertyName: _classname
 
    PensioenAanspraakType:
      type: string
      enum:
        - TYPE_A
        - TYPE_B






```

**Comment by alevinetx** [comment]
@robinbakkerus  : Did you find an answer for this?   I did find this package that contains the class, and looks like it may be what was intended:

https://pub.dev/packages/codable

**Comment by donkii** [comment]
Is there any solution yet? We are having the same problem.

**Comment by wing328** [comment]
Dart client does not support oneOf at the moment. We definitely welcome contribution on that.

What does the correct code look like?

**Comment by johnsogg** [comment]
Is `oneOf` still missing for the dart client? 

I'm tracking a problem where our OpenApi spec includes this for a request schema:

```
              oneOf:
                - $ref: '#/components/schemas/FirstThing'
                - $ref: '#/components/schemas/SecondThing'
```

... and the generated Dart code is confused:

```
Future<Response> someCallWithHttpInfo(UNKNOWN_BASE_TYPE UNKNOWN_BASE_TYPE)
```

If I replace the `oneOf` with a single reference, all is fine. 

Except we really do need to support multiple possible input schemas. The workaround, I think, is to just declare different paths based on the input schemas.

**Comment by jamesblasco** [comment]
What about if we use mixin for generating oneOf? From the example above 

```
 aanspraken:
          type: array
          items:
            oneOf:
              - $ref: '#/components/schemas/VroegPensioenAanspraak'
              - $ref: '#/components/schemas/WeduwePensioenAanspraak'
            discriminator:
              propertyName: _classname
```

```dart
mixin AansprakenOneOf {}

class VroegPensioenAanspraak implements AansprakenOneOf {}

class WeduwePensioenAanspraak implements AansprakenOneOf {}
```

**Comment by wing328** [comment]
This should be fixed in v6.0.0. Please pull the latest master to give it another try.
